src/core/iomgr/pollset_windows.c

```diff
@@ -55,7 +56,10 @@ void grpc_pollset_init(grpc_pollset *pollset) {
 void grpc_pollset_shutdown(grpc_pollset *pollset,
                            void (*shutdown_done)(void *arg),
                            void *shutdown_done_arg) {
-  grpc_pollset_kick(pollset);
+  gpr_mu_lock(&pollset->mu);
+  pollset->shutting_down = 1;
+  gpr_mu_unlock(&pollset->mu);
+  gpr_cv_broadcast(&pollset->cv);
```

Comment:
Better to broadcast inside of the lock (leads to more predictable scheduling in some cases, and this is actually critical path right now).